src/traces/heatmap/plot.js

```diff
@@ -337,7 +337,14 @@ function plotOne(gd, plotinfo, cd) {
 
     if(zsmooth) { // best or fast, works fastest with imageData
         var pxIndex = 0,
+            pixels;
+
+        try {
             pixels = new Uint8Array(imageWidth * imageHeight * 4);
```

Comment:
> what if we search for another polyfill that handles > 1e5 arrays?

In order to *really* make something that behaves like a typed array, you need to add a whole lot of machinery, so no matter what you do it's going to be much slower than a bare array. That might be justified if you truly need to enforce the type restrictions, but we don't, we already have constructed our colors to be integers 0-255. So this overhead is simply harmful to us, far better to use a bare `Array` than any conceivable polyfill.

So then the question is, is there *ever* a situation where we truly need the polyfill? Seems like no, right? Because any system that supports WebGL will already support typed arrays, and nothing else is affected by this.

And then I guess we could ask if there's a direct way to tell if we have *real* typed arrays vs polyfills. But honestly, I think based on my perf testing above, for the modern case - where we don't hit either of those `catch` blocks - this is actually the fastest possible code; short of doing the feature detection at compile time and completely swapping out the `plot` routine 😱  we're not going to get less overhead than this. And for the fallback case we're only talking about a few tens of µs delay, hardly problematic for folks whose browsing experience is likely subject to far worse restrictions and bugs by now.